src/components/shows/SawItSwitch.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState, useEffect, useCallback, useContext } from 'react';
+import { IUser } from '../users/Users';
+import { Switch } from '@material-ui/core';
+import axios from 'axios'
+import { AppContext } from '../../context/AppProvider';
+
+interface Props {
+    showId: string
+    currentUser: IUser
+}
+
+const SawItSwitch: React.FC<Props> = ({ showId, currentUser }) => {
+    const [sawIt, setSawIt] = useState(false)
+    const {state} = useContext(AppContext);
+    const {attendances} = state
+
+    const handleSawItChange = () => (event: React.ChangeEvent<HTMLInputElement>) => {
+        postSawIt(event.target.checked)
+    };
+
+    const postSawIt = useCallback(async (value) => {
+        await axios({
+            method: 'post',
+            url: `${process.env.REACT_APP_API_URL}/shows/${showId}/${value ? "attend" : "unattend"}`,
+            headers: {
+                "Content-Type": "multipart/form-data",
+                "Authorization": currentUser?.token
+            }
+        });
+        setSawIt(value)
+
+        // need to update the attendances in state from here
```

Comment:
@shaunoff i'm strugging with how to update the attendances array in the app context state.  the examples that i'm seeing have the app provider expose state and setState, but the dispatch/reducer stuff - not quite sure how that fits in.